### DISCUSSION TITLE
Huge performance regression somewhere between v1.10.0 and v1.10.3

### DISCUSSION THREAD

**Discussion by LoiLock** [pmndrs/valtio#715] [General]
In our Next.js project, upgrading to v1.10.3 caused a rendering time of about 6 whole seconds, during which the whole site is unresponsive, in v1.10.0 it's ~100-200ms.

We're storing a pretty large response object in the valtio store (~148kb of json). This object contains anywhere between 2000 and 4000 values, all of which are individually accessed in child components. The keys to access those values are assembled from other values also found in this response, so we can expect ~10-15k values being obtained from the store during rendering (all calls being made separately).
I saw that there were some performance fixes released between these versions, could we potentially make these fixes optional?

v1.10.1 makes it a lot worse compared to v1.10.0
v1.10.2 and v1.10.3 are a little bit worse than v1.10.1

**Comment by dai-shi** [comment]
Could you identify which PR causes it?
We'd like to understand the issue and then consider some workarounds.
